File: Omega_v7_2/OmegaBridge/node/worker.js
const fs = require("fs");

function generateSignal() {
  return {
    type: "node_worker_signal",
    from: "worker_node",
    to: "python_core",
    payload: {
      action: "learn",
      value: Math.random()
    },
    timestamp: Date.now()
  };
}

setInterval(() => {
  const msg = generateSignal();

  fs.writeFileSync("../bus/omega_message.json", JSON.stringify(msg, null, 2));
  console.log("⚙️ Worker generated signal");
}, 3000);

File: Omega_v7_2/OmegaBridge/python/worker.py
import json
import time
import random

FILE = "../bus/omega_message.json"

while True:
    msg = {
        "type": "python_worker_signal",
        "from": "python_worker",
        "to": "node_core",
        "payload": {
            "confidence": random.random(),
            "decision": random.choice(["explore", "exploit"])
        },
        "timestamp": time.time()
    }

    with open(FILE, "w") as f:
        json.dump(msg, f, indent=2)

    print("🐍 Python worker sent signal")
    time.sleep(4)
